src/providers/oauth-provider.js

```diff
@@ -181,9 +181,20 @@ function OAuthProvider() {
        */
 
       revokeToken() {
-        var data = queryString.stringify({
+        var data = {
+          client_id: config.clientId,
+          client_secret: config.clientSecret,
           token: OAuthToken.getRefreshToken() ? OAuthToken.getRefreshToken() : OAuthToken.getAccessToken()
-        });
+        };
+
+        // To remove null values
+        for (var index in data) {
```

Comment:
Ahm. You changed the latest code I reviewed.

Sorry, but can you keep the logic used in the other methods, i.e.,

```
var data = {
  client_id: config.clientId,
  token: OAuthToken.getRefreshToken() ? OAuthToken.getRefreshToken() : OAuthToken.getAccessToken()
};

if (null !== config.clientSecret) {
  data.client_secret = config.clientSecret;
}

data = queryString.stringify(data);
```